Fix predispatch tracing for aten::lift_fresh_copy

tugsbayasgalan opened the pull request:
Differential Revision: D56200666


Previously, when we hit the Functionalize kernel for lift_fresh_copy, we directly dispatch self.clone() to proxy dispatch. As a result, we end up receiving a functional tensor at proxy dispatch. As a work around, I unwrap self manually. Not sure, why it works ok in aot-dispatch tho

bdhirsh commented on aten/src/ATen/FunctionalizeFallbackKernel.cpp:
```diff
@@ -210,7 +210,13 @@ static at::Tensor lift_fresh_functionalize_copy(const at::Tensor & self) {
+    // When we are tracing under PreDispatch, PreDispatch key will be
+    // in the local include TLS. As a result, when we redispatch here,
+    // we will end up hitting PreDispatch stack first. So, we should
+    // directly redispatch to the functionalize key manually.
```
nit: if we have a comment about why we call `._op_dk(DispatchKey.Functionalize)` from FunctionalTensorMode, it'd be good to link to that comment and mention the underlying reason is the same

tugsbayasgalan committed:
Fix predispatch tracing for aten::lift_fresh_copy (#124198)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/124198

Test Plan: CI

Reviewed By: bdhirsh

Differential Revision: D56200666

tugsbayasgalan commented:
@pytorchbot merge -f "Landed internally"
